typescript/api/services/TriggerService.ts

```diff
@@ -185,68 +160,20 @@ export module Services {
-      //optional
-      let fieldLanguageCode = _.get(options, 'fieldLanguageCode');
-      let arrayObjFieldDBName = _.get(options, 'arrayObjFieldDBName');
-      //Set false by default if not present this option will remove leading and trailing spaces from a none array value
-      //then it will modify the value in the record if the the regex validation is passed therefore handle with care
-      let trimLeadingAndTrailingSpacesBeforeValidation = _.get(options, 'trimLeadingAndTrailingSpacesBeforeValidation') || false;
-      let caseSensitive = _.isUndefined(_.get(options, 'caseSensitive', undefined)) ? true : _.get(options, 'caseSensitive', undefined); // default to true
-      let allowNulls = _.isUndefined(_.get(options, 'caseSensitive', undefined)) ? true : _.get(options, 'caseSensitive', undefined); // default to true for backwards compatibility
-
-      let data = _.get(record, fieldDBName);
-      // Fail fast if the field is empty and allowNulls is false
-      if (_.isEmpty(data) && !allowNulls) {
-        let customError: RBValidationError;
-        if (!_.isUndefined(fieldLanguageCode)) {
-          let fieldName = TranslationService.t(fieldLanguageCode);
-          let baseErrorMessage = TranslationService.t(errorMessageCode);
-          customError = new RBValidationError(fieldName + ' ' + baseErrorMessage);
-        } else {
-          let baseErrorMessage = TranslationService.t(errorMessageCode);
-          customError = new RBValidationError(baseErrorMessage);
-        }
-        throw customError;
-      }
-
-      if (_.isArray(data) && !_.isUndefined(arrayObjFieldDBName)) {
-
-        sails.log.verbose(`validateFieldUsingRegex is array ${fieldDBName} ` + JSON.stringify(data));
-        sails.log.verbose('validateFieldUsingRegex is array regexPattern ' + regexPattern);
-        for (let row of data) {
-
-          let objField = _.get(row, arrayObjFieldDBName);
-
-          if (!_.isUndefined(objField) && objField != null && objField != '' && !_.isUndefined(regexPattern) && !_.isUndefined(errorMessageCode)) {
-
-            this.validateRegex(objField, regexPattern, fieldLanguageCode, errorMessageCode, caseSensitive);
-          }
-        }
-
-      } else {
-
-        sails.log.verbose(`validateFieldUsingRegex ${fieldDBName} ` + data);
-        sails.log.verbose('validateFieldUsingRegex regexPattern ' + regexPattern);
-        if (!_.isUndefined(data) && data != null && data != '' && !_.isUndefined(regexPattern) && !_.isUndefined(errorMessageCode)) {
-
-          if (trimLeadingAndTrailingSpacesBeforeValidation) {
-            let trimData = _.trim(data);
-            data = trimData;
-          }
-
-          this.validateRegex(data, regexPattern, fieldLanguageCode, errorMessageCode, caseSensitive);
-
-          if (trimLeadingAndTrailingSpacesBeforeValidation) {
-            _.set(record, fieldDBName, data);
-          }
-        }
-      }
-
-      return record;
+      return await sails.helpers.validateRegexp.with({
```

Comment:
Can this part be rolled back? We're aware of helpers and have made a conscious decision not to use them at this stage.

Reply:
Ok. Could the decision to not user sails helpers please be documented somewhere? Maybe a `readme.md` file in the sails `api/helpers` folder?

Reply:
Yep good idea